packages/medusa/src/services/product.js

```diff
@@ -293,6 +299,10 @@ class ProductService extends BaseService {
         rest.discountable = false
       }
 
+      if (rest.variants)
+        for (const [i, variant] of rest.variants.entries())
+          variant.variant_rank = i
+
```

Comment:
variants are not created as part of the `create` call here - this happens in the route which you are already accounting for so please delete this.

Reply:
I removed it